src/opera/pge/base/runconfig.py

```diff
@@ -252,6 +280,27 @@ def sas_schema_path(self) -> str:
             else resource_filename('opera', sas_schema_path)
         )
 
+    @property
+    def algorithm_parameters_schema_path(self) -> str:
+        """Returns the path to the algorithm parameter Schema file for DSWX-S1"""
+        algorithm_parameters_schema_path = self._pge_config['PrimaryExecutable']['AlgorithmParametersSchemaPath']
+        return (
+            algorithm_parameters_schema_path
+            if isabs(algorithm_parameters_schema_path)
+            else resource_filename('opera', algorithm_parameters_schema_path)
+        )
+
+    @property
+    def algorithm_parameters_config_path(self) -> str:
+        """Returns the path to the optional algorithm_parameter run configuration file for DSWX-S1"""
+        algorithm_parameters_config_path = \
+            self._sas_config['runconfig']['groups']['dynamic_ancillary_file_group']['algorithm_parameters']
```

Comment:
Since not all SAS utilize a separate algorithm parameters run config, this method needs to be a bit sensitive to that. 

We should wrap this function body with a try/except block that looks for `KeyError` (i.e. when there is no `algorithm_parameters` field in the parsed runconfig), and if we hit the except block we can assume the field is not present and just return `None` instead.